### PR TITLE
Update to be Rails 6.1 compatible

### PR DESCRIPTION
**Pull request by amy-hs**
Changes include:
- Adding Rails 6.1 compatibility
  - Including refactoring of the `config.rb` file
- Updating Ruby version to 2.7.8
- Updating the Ruby matrix since we have dropped support for Ruby 2.4 and 2.5
- Version bump to 1.4.0

Co-authors: @amy-hs @davkutalek

### CODE REVIEW

File: lib/knockoff/config.rb [hunk 1]
```diff
@@ -34,7 +34,10 @@ def replica_env_keys
     end
 
     def update_replica_configs(new_configs)
-      ActiveRecord::Base.configurations['knockoff_replicas'].merge(new_configs) if ActiveRecord::Base.configurations['knockoff_replicas'].present?
+      if ActiveRecord::Base.configurations.configs_for(env_name: 'knockoff_replicas').present?
+        ActiveRecord::Base.configurations.configs_for(env_name: 'knockoff_replicas').first.configuration_hash.merge(new_configs)
```
Comment (soupmatt) [inline lib/knockoff/config.rb:38]: Does this work? `Hash#merge` doesn't modify the receiver, but returns a modified copy. It looks like `configuration_hash` returns a hash that has been frozen. (see [source](https://github.com/rails/rails/blob/v6.1.7.6/activerecord/lib/active_record/database_configurations/hash_config.rb#L31)) 

I believe that means this line isn't doing anything.

Is it possible that we are wanting something like one of these?

```ruby
new_configs.symbolize_keys!.merge!(ActiveRecord::Base.configurations.configs_for(env_name: 'knockoff_replicas').first.configuration_hash)
```

or

```ruby
new_configs.symbolize_keys!.reverse_merge!(ActiveRecord::Base.configurations.configs_for(env_name: 'knockoff_replicas').first.configuration_hash)
```

**Review by soupmatt — CHANGES_REQUESTED** [review]
I'm unsure about the merging of the `knockoff_replicas` config values. See my comments in the diff.

File: lib/knockoff/config.rb (hunk 1 above)
Comment (seanmsmith23) [reply lib/knockoff/config.rb:38]: Yeah... interesting. It did the same thing previously, has this ever done anything?

Comment (soupmatt) [reply lib/knockoff/config.rb:38]: Agreed that it didn't do anything before. In fact, it looks like the `knockoff_replicas` stanza in `config/database.yml` is basically ignored.

Comment (amy-hs) [reply lib/knockoff/config.rb:38]: Do you think we should remove this logic?

Comment (amy-hs) [reply lib/knockoff/config.rb:38]: This is where we call the method: https://github.com/joinhandshake/knockoff/blob/main/lib/knockoff.rb#L43

And ultimately, it gets used [here](https://github.com/joinhandshake/handshake/blob/main/packs/elasticsearch_indexer/app/lib/elasticsearch_indexer/index_creator.rb#L41).

Comment (soupmatt) [reply lib/knockoff/config.rb:38]: Either remove it or fix it so it does something, but I'm unsure which is best.

I'm going to bring it up in our staff eng channel and see what others think.

Comment (davkutalek) [reply lib/knockoff/config.rb:38]: merge is causing an error to be thrown when running dev:fast_reset on the 6.1 branch

```NoMethodError: undefined method `merge' for #<ActiveRecord::DatabaseConfigurations::HashConfig:0x00000001069f1d50 @env_name="knockoff_replicas", @name="primary", @configuration_hash={:adapter=>"postgresql", :min_messages=>"warning", :username=>nil, :password=>nil, :prepared_statements=>false, :reaping_frequency=>10, :pool=>10, :variables=>{"lock_timeout"=>nil, "statement_timeout"=>nil}}>```

Comment (amy-hs) [reply lib/knockoff/config.rb]: I merged in @davkutalek's changes into this branch 👍

**Review by soupmatt — APPROVED** [review]
LGTM

**Review by Mctalian — APPROVED** [review]

**Pull request merged**